src/Tezos/Forging.h

```diff
@@ -8,14 +8,64 @@
 
 #include "../PublicKey.h"
 #include "../proto/Tezos.pb.h"
+#include "uint256.h"
 
 #include <string>
+#include <vector>
+#include <variant>
 
 using namespace TW;
 using namespace TW::Tezos::Proto;
 
+enum PrimType : std::uint8_t {
+    Pair = 7,
+};
+
+struct MichelsonValue;
+
+struct PrimValue {
+    std::string prim;
+    std::vector<MichelsonValue> args;
+    std::vector<std::string> anots;
+};
+
+struct BytesValue {
+    std::string bytes;
+};
+
+struct StringValue {
+    std::string string;
+};
+
+struct IntValue {
+    std::string _int;
+};
+
+struct MichelsonValue {
+    using MichelsonArray = std::vector<std::variant<PrimValue, BytesValue, StringValue, IntValue>>;
+    using MichelsonVariant = std::variant<
+        PrimValue,
+        BytesValue,
+        StringValue,
+        IntValue,
+        MichelsonArray>;
+    MichelsonVariant value;
+};
+
+
+MichelsonValue::MichelsonVariant FA12ParameterToMichelson(const FA12TransactionOperationData& data);
+MichelsonValue::MichelsonVariant FA2ParameterToMichelson(const FA2TransactionOperationData& data);
```

Comment:
Should we create a new file? `Michelson.cpp/h` ?

Reply:
Yeah, let's do it